WordPress/gutenberg
# Refactor useMediaQuery with useSyncExternalStore

## What?

PR refactors `useMediaQuery` hook to use `useSyncExternalStore` instead of `useEffect`.

## Why?
The `useSyncExternalStore` is better suited for similar subscriptions.

## Testing Instructions
1. Open a post.
2. Resize the browser window, and confirm panels adjust to the width as before.
3. Repeat the same step in the Site Editor.

## Screenshots or screencast 

https://user-images.githubusercontent.com/240569/225267644-db196bc7-fb1d-4728-8ebb-1d370403bc9d.mp4
